src/VecSim/spaces/IP_space.cpp

```diff
@@ -30,21 +30,21 @@ dist_func_t<float> IP_FP32_GetDistFunc(size_t dim, const Arch_Optimization arch_
 
     switch (arch_opt) {
     case ARCH_OPT_AVX512_F:
-#ifdef __AVX512F__
+#ifdef OPT_AVX512F
```

Comment:
why not use the compiler's built-in flags?

Reply:
because we actually don't want to compile this file with the optimization flags, even though it worked until now. We don't want an unsupported instruction to slip in and cause a crash on non-supporting machines.
Using our own indication of whether we compiled the optimizations seems clean. Can we turn on these flags without using the optimizations (`-mavx`)? Is it good practice to `define` these flags without using the optimizations?

Reply:
Thanks. i agree this is a better practice.